modules/packages/Sort.chpl

```diff
@@ -802,6 +873,390 @@ proc selectionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
     compilerError("selectionSort() requires 1-D array");
 }
 
+pragma "no doc"
+proc shellSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator,
+               start=Dom.low, end=Dom.high)
+{
+  chpl_check_comparator(comparator, eltType);
+
+  if Dom.rank != 1 then
+    compilerError("shellSort() requires 1-D array");
+  if Dom.stridable then
+    compilerError("shellSort() requires an array over a non-strideable domain");
+
+  // Based on Sedgewick's Shell Sort -- see
+  // Analysis of Shellsort and Related Algorithms 1996
+  // and see Marcin Ciura - Best Increments for the Average Case of Shellsort
+  // for the choice of these increments.
+  var n = 1 + end - start;
+  var js,hs:int;
+  var v,tmp:Data.eltType;
+  const incs = (701, 301, 132, 57, 23, 10, 4, 1);
+  const nincs = incs.size;
+  for h in incs {
+    hs = h + start;
+    for is in hs..end {
+      v = Data[is];
+      js = is;
+      while js >= hs && chpl_compare(v,Data[js-h],comparator) < 0 {
+        Data[js] = Data[js - h];
+        js -= h;
+      }
+      Data[js] = v;
+    }
+  }
+}
+
+// This is the number of bits to sort at a time in the radix sorter.
+// The code assumes that all integer types are a multiple of it.
+// That would need to change if it were to increase.
+//
+// At the same time, using a value less than 8 will probably perform poorly.
+private param RADIX_BITS = 8;
+
+// This structure tracks configuration for the radix sorter.
+pragma "no doc"
+record MSBRadixSortSettings {
```

Comment:
I still find this odd compared to other interfaces, but if it's not part of the public interface for this release, we can leave it as is for now.